src/HybridSystems.jl

```diff
@@ -100,19 +107,26 @@ Returns the target mode for the transition `t`.
 """
 target_mode(hs::HybridSystem, t) = hs.modes[target(hs, t)]
 
+"""
+    resetmap(hs::HybridSystem, t)
+
+Returns the reset map for the transition `t`.
+"""
+resetmap(hs::HybridSystem, t) = hs.resetmaps[symbol(hs, t)]
+
 """
     assignment(hs::HybridSystem, t)
 
 Returns the assignment for the transition `t`.
 """
-assignment(hs::HybridSystem, t) = hs.resetmaps[symbol(hs, t)]
+assignment(hs::HybridSystem{A,S,R,W}, t) where {A, S<:AbstractSystem, R<:AbstractMap, W} = args -> apply(resetmap(hs, t), args)
```

Comment:
Should it be `args...` ?

Reply:
Like `args... -> apply(resetmap(hs, t), args...)`? Is that valid syntax?

Reply:
`(args...) -> apply(resetmap(hs, t), args...)` should be valid

Reply:
> `(args...) -> apply(resetmap(hs, t), args...)` should be valid

👍  That one allows for either `x` as vector and also both `x, u`:

```julia
struct MyAffineMap
    A::Matrix{Float64}
    b::Vector{Float64}
end

apply(t::MyAffineMap, x) = t.A * x + t.b

assignment(m) = args -> apply(m, args)
assignment_splat(m) = args -> apply(m, args...)
assignment_splat_v2(m) = (args...) -> apply(m, args...)

m = MyAffineMap(rand(2, 2), rand(2))
MyAffineMap([0.063433 0.0320108; 0.845872 0.708518], [0.505035, 0.343215])

x = rand(2)
a = assignment(m)(x)
2-element Array{Float64,1}:
 0.5683784908782451
 1.4186773435138202

a = assignment_splat(m)(x)
MethodError: no method matching apply(::MyAffineMap, ::Float64, ::Float64)
Closest candidates are:
  apply(::MyAffineMap, ::Any) at In[1]:6

a = assignment_splat_v2(m)(x)
2-element Array{Float64,1}:
 0.5683784908782451
 1.4186773435138202
```

But:

```julia
struct MyAffineControlMap
    A::Matrix{Float64}
    B::Matrix{Float64}
end

apply(t::MyAffineControlMap, x, u) = t.A * x + t.B * u

m = MyAffineControlMap(rand(2, 2), rand(2,2))
MyAffineControlMap([0.36716 0.709987; 0.393566 0.925329], [0.863631 0.316295; 0.356386 0.118833])

u = rand(2);

a = assignment(m)(x, u)
MethodError: no method matching (::getfield(Main, Symbol("##3#4")){MyAffineControlMap})(::Array{Float64,1}, ::Array{Float64,1})
Closest candidates are:
  #3(::Any) at In[2]:1

a = assignment_splat(m)(x, u)
MethodError: no method matching (::getfield(Main, Symbol("##5#6")){MyAffineControlMap})(::Array{Float64,1}, ::Array{Float64,1})
Closest candidates are:
  #5(::Any) at In[2]:2

a = assignment_splat_v2(m)(x, u)
2-element Array{Float64,1}:
 1.6265315910625637
 1.3263235565255767
```